urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/agent/infocollect/DynamicCPUInfo.java

```diff
@@ -0,0 +1,226 @@
+	 * 将从stat文件取出来的每一行数据复制给mind[]数组
+	 * 
+	 * @param str
+	 *            读文件的每一行数据
+	 * @return mind数组
+	 */
+	private double[] getArray(String str) {
+		double mind[] = new double[5];
+		String a[] = str.trim().split(" ");
+		switch (a.length) {
+		case 5: // 兼容Red Hat 9版本，内核，2.4.20
+			mind[0] = Double.parseDouble(a[1]);
+			mind[1] = Double.parseDouble(a[2]);
+			mind[2] = Double.parseDouble(a[3]);
+			mind[3] = Double.parseDouble(a[4]);
+			mind[4] = 0;
+			break;
+		case 8: // 兼容Red Hat Enterprise Linux AS release 4版本，/proc/stat中的数字个数为7
+			mind[0] = Double.parseDouble(a[1]) + Double.parseDouble(a[7]);
+			mind[1] = Double.parseDouble(a[2]);
+			mind[2] = Double.parseDouble(a[3]) + Double.parseDouble(a[6]);
+			mind[3] = Double.parseDouble(a[4]);
+			mind[4] = Double.parseDouble(a[5]);
+			break;
+		case 9: // 兼容Red Hat Enterprise Linux Server release
+			// 5.2版本,/proc/stat中的数字个数为8
+			mind[0] = Double.parseDouble(a[1]) + Double.parseDouble(a[7]);
+			mind[1] = Double.parseDouble(a[2]);
+			mind[2] = Double.parseDouble(a[3]) + Double.parseDouble(a[6]) + Double.parseDouble(a[8]);
+			mind[3] = Double.parseDouble(a[4]);
+			mind[4] = Double.parseDouble(a[5]);
+			break;
+		case 10: // 兼容Red Hat Enterprise Linux Server release
+			// 6.0版本,/proc/stat中的数字个数为9
+			mind[0] = Double.parseDouble(a[1]) + Double.parseDouble(a[7]) + Double.parseDouble(a[9]);
+			mind[1] = Double.parseDouble(a[2]);
+			mind[2] = Double.parseDouble(a[3]) + Double.parseDouble(a[6]) + Double.parseDouble(a[8]);
+			mind[3] = Double.parseDouble(a[4]);
+			mind[4] = Double.parseDouble(a[5]);
+			break;
+		default:
+			mind[0] = Double.parseDouble(a[1]);
+			mind[1] = Double.parseDouble(a[2]);
+			mind[2] = Double.parseDouble(a[3]);
+			mind[3] = Double.parseDouble(a[4]);
+			mind[4] = Double.parseDouble(a[5]);
+			;
+		}
+		return mind;
+	}
+
+	/**
+	 * 从stat文件获取信息，返回的是map结构
+	 * 
+	 * @return cpuInfoMap
+	 * @throws IOException
+	 */
+	private Map<String, double[]> getInfoFromStat() throws IOException {
+		Map<String, double[]> cpuInfoMap = new HashMap<String, double[]>();
+		FileReader reader = null;
+		BufferedReader br = null;
+		try {
+			reader = new FileReader("/proc/stat");
+			br = new BufferedReader(reader);
+			String str;
+			while ((str = br.readLine()) != null) {
+				if (str.contains("cpu")) {
+					str = str.replaceAll(" {2,}", " ");
+					String a[] = str.split(" ");
+					double[] buffer = getArray(str);
+					cpuInfoMap.put(a[0], buffer);
+				}
+			}
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			LOG.error(LogUtils.getTrace(e));
+		} finally {
+			if (br != null)
+				br.close();
+			if (reader != null)
+				reader.close();
+		}
+		return cpuInfoMap;
+	}
+
+	@Override
+	public void run() {
+		// To change body of implemented methods use File | Settings | File
+		// Templates.
+		while (running) {
+			try {
+				convertToCPUInfo();
+				try {
+					Thread.sleep(samplingInterval);
+				} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)